routers/common/middleware.go

```diff
@@ -15,24 +15,23 @@ import (
 	"code.gitea.io/gitea/modules/setting"
 	"code.gitea.io/gitea/modules/web/routing"
 
+	"gitea.com/go-chi/session"
 	"github.com/chi-middleware/proxy"
 	chi "github.com/go-chi/chi/v5"
 )
 
-// Middlewares returns common middlewares
-func Middlewares() []func(http.Handler) http.Handler {
-	handlers := []func(http.Handler) http.Handler{
-		func(next http.Handler) http.Handler {
-			return http.HandlerFunc(func(resp http.ResponseWriter, req *http.Request) {
-				// First of all escape the URL RawPath to ensure that all routing is done using a correctly escaped URL
-				req.URL.RawPath = req.URL.EscapedPath()
+// ProtocolMiddlewares returns HTTP protocol related middlewares
+func ProtocolMiddlewares() (handlers []any) {
```

Comment:
Why we mention HTTP protocol? I think all these middlewares are based on HTTP?

Reply:
"HTTP protocol related"

Others are for application logic (eg: auth, session), not for base HTTP protocol logic.